dev/pastetools/getclipboard.html

```diff
@@ -0,0 +1,309 @@
+<!DOCTYPE html>
+<!--
+Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+-->
+<html>
+<head>
+	<meta charset="UTF-8">
+	<title>Get Clipboard HTML and RTF</title>
+	<style>
+
+		textarea {
+			border: 1px solid #808080;
+			float: left;
+			height: 200px;
+			width: 100%;
+			overflow: auto;
+			margin-bottom: 20px;;
+		}
+		label {
+			display: block;
+			line-height: 2;
+		}
+		span {
+			background-color: #88FFAA;
+		}
+		.marker {
+			background-color: #FF9999;
+		}
+	</style>
+	<script src="../../ckeditor.js"></script>
+</head>
+<body>
+	<p>Paste Inside the Editor:</p>
+	<div id="editor">
+		<textarea id="input"></textarea>
+	</div>
+	<div>
+		<div style="float: left; width: 49%">
+			<p>Raw HTML Data Received: <span id="emHtml"></span></p>
+			<textarea data-name="input.html" id="rawHtml" readonly="readonly"></textarea>
+			<button id="htmlData">Save HTML Data</button>
+		</div>
+		<div style="float: right; width: 49%">
+			<p>Raw RTF Data Received: <span id="emRtf"></span></p>
+			<textarea data-name="input.rtf" id="rawRtf" readonly="readonly"></textarea>
+			<button id="rtfData">Save RTF Data</button>
+		</div>
+	</div>
+	<div style="width: 100%; float: left;">
+		<p>After Paste Processing:</p>
+		<textarea id="output" readonly="readonly"></textarea>
+	</div>
+
+	<div id="configuration">
+		<p>Editor configuration:</p>
+
+		<div>
+			<button id="update">Update editor</button>
+		</div>
+		<div id="acf">
+			<label><input type="checkbox" id="allowedContent"> Allow all content</label>
+			<label>Extra allowed content: <input type="text" id="extraAllowedContent"></label>
+		</div>
+		<div id="editable"></div>
+		<div id="paste"></div>
+		<div id="basic"></div>
+		<div id="image"></div>
+		<div id="other"></div>
+	</div>
+
+	<script>
+	var editablePlugins = [
```

Comment:
I'm wondering if there is a better method to list plugins. They are nicely labeled which is a +, but we will need to update this file every time when a new plugin has been added. Theoretically, we could just list plugins inside `plugins/*` folder, because this file uses repository editor instance. Not sure if it's worth the effort though, and we will lose plugins sections. WDYT?

Reply:
That was my first approach. I prepared simple Node.js script that just iterated through all plugins and generate HTML code from it… But there was too much noise, e.g. `button`, `panel` etc. Unfortunately we have too many abstract plugins to use such approach. As I had to filter plugins, I've decided to just do it _my way_ ;)

Reply:
Make sense to me :+1: